nx/lib/nx/binary_backend.ex

```diff
@@ -737,6 +746,23 @@ defmodule Nx.BinaryBackend do
   defp element_logical_xor(_, _, r) when r == 0, do: 1
   defp element_logical_xor(_, _, _), do: 0
 
+  defp validate_numbers(a, b, fun) do
```

Comment:
This validation should not be here. It should happen in the `Nx` module. Those functions should fail early if they get a complex number.